File: examples/kotlin/Sender.kt
// --- Sender ---
// (Compile with kotlinc Sender.kt -include-runtime -d sender.jar)
import java.net.DatagramPacket
import java.net.DatagramSocket
import java.net.InetAddress

fun main() {
    val socket = DatagramSocket()
    val serverAddress = InetAddress.getByName("localhost")
    val message = "Hello, URP!".toByteArray()

    println("Sending message to server...")
    val packet = DatagramPacket(message, message.size, serverAddress, 8080)
    socket.send(packet)

    val buffer = ByteArray(1024)
    val responsePacket = DatagramPacket(buffer, buffer.size)
    socket.receive(responsePacket)
    val response = String(responsePacket.data, 0, responsePacket.length)
    println("Received response: '$response'")
    socket.close()
}

File: examples/javascript/sender.js
// --- Sender ---
const dgram = require('dgram');
const client = dgram.createSocket('udp4');

console.log('Sending message to server...');

const message = Buffer.from('Hello, URP!');
client.send(message, 8080, 'localhost', (err) => {
  if (err) {
    console.error('Error sending message:', err);
    client.close();
  }
});

client.on('message', (msg, rinfo) => {
  console.log(`Received response: '${msg}'`);
  client.close();
});

client.on('error', (err) => {
  console.error('Socket error:', err);
  client.close();
});

// Timeout after 5 seconds
setTimeout(() => {
  console.log('Timeout: No response received');
  client.close();
}, 5000);
